server/types/array.go

```diff
@@ -148,6 +148,18 @@ func (ac arrayContainer) Convert(val any) (any, sql.ConvertInRange, error) {
 	switch val := val.(type) {
 	case []any:
 		return val, sql.InRange, nil
+	case []int16:
+		anyArray := make([]any, len(val))
+		for i, s := range val {
+			anyArray[i] = s
+		}
+		return anyArray, sql.InRange, nil
+	case []uint32:
+		anyArray := make([]any, len(val))
+		for i, s := range val {
+			anyArray[i] = s
+		}
+		return anyArray, sql.InRange, nil
```

Comment:
All array types **must** be `[]any`. Also, `Convert` shouldn't have any additional logic in it. We can't quite delete the function due to how GMS uses it, so instead we treat it as a gatekeeper that just checks the type and nothing else.